R/slide.R

```diff
@@ -734,27 +875,9 @@ epi_slide_opt <- function(
       arrange(.data$time_value)
 
     if (f_from_package == "data.table") {
-      # If a group contains duplicate time values, `frollmean` will still only
-      # use the last `k` obs. It isn't looking at dates, it just goes in row
-      # order. So if the computation is aggregating across multiple obs for the
-      # same date, `epi_slide_opt` and derivates will produce incorrect results;
-      # `epi_slide` should be used instead.
-      if (anyDuplicated(.data_group$time_value) != 0L) {
-        cli_abort(
-          c(
-            "group contains duplicate time values. Using `epi_slide_[opt/mean/sum]` on this
-              group will result in incorrect results",
-            "i" = "Please change the grouping structure of the input data so that
-              each group has non-duplicate time values (e.g. `x %>% group_by(geo_value)
-              %>% epi_slide_opt(.f = frollmean)`)",
-            "i" = "Use `epi_slide` to aggregate across groups"
-          ),
-          class = "epiprocess__epi_slide_opt__duplicate_time_values",
-          epiprocess__data_group = .data_group,
-          epiprocess__group_key = .group_key
-        )
-      }
-
+      # Grouping should ensure that we don't have duplicate time values.
```

Comment:
Ooh, right, it currently doesn't completely ensure it.  Dmitry tested out placing a check in `dplyr_reconstruct` but it was too slow for how often it runs.  I've looked into a couple of ways to address this:
- Fixing our `dplyr_extending` usage which is pretty wonky, so we wouldn't run checks as often.  (experimentation [here](https://github.com/brookslogan/dplyr.extending.test/), incomplete)
- Making the check faster (#560) 
I didn't complete those, so

todo: add back duplicated time values check for now

`new_epi_df` shouldn't validate something complicated like this if we try to follow [Hadley recommendations](https://adv-r.hadley.nz/s3.html).  Related: https://github.com/cmu-delphi/epiprocess/issues/312.

I'm not sure about not exporting `new_epi_df`; https://github.com/cmu-delphi/epiprocess/issues/94 suggests it's needed downstream, but I don't actually see it anywhere in epipredict right now.  We should probably at least provide a helper `epi_df` that acts like `tibble`, which would be convenient and make `new_epi_df` not stand out as much.  Filed off into #572.

Reply:
Shouldn't the duplicate time value check also exist for the slider functions?  We're not using `slide_index_*` so I'd expect it's similar to the `data.table::froll*`.  (And `slide_index*` would just include everything in the time window + produce duplicate outputs rather than abort...)

Reply:
The duplicate check used to be applied to both `data.table` and `slider` cases. It was moved to only the `data.table` case [here](https://github.com/cmu-delphi/epiprocess/commit/1fdafb2df2a5f984632593b566c10fb9dc994b24), but I'm assuming that was an accident. Any insight, @dshemetov?

Reply:
Brought duplicate check back, but for all slide functions.  I also applied the apparently faster key-uniqueness check from #560.  This could use a look over.